Code/GraphMol/MolEnumerator/LinkNode.h

```diff
@@ -0,0 +1,131 @@
+//
+//  Copyright (C) 2020 Greg Landrum and T5 Informatics GmbH
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#include <RDGeneral/Invariant.h>
+
+#include <boost/lexical_cast.hpp>
+#include <boost/tokenizer.hpp>
+#include <boost/format.hpp>
+#include <algorithm>
+
+typedef boost::tokenizer<boost::char_separator<char>> tokenizer;
+
+namespace RDKit {
+namespace MolEnumerator {
+
+struct LinkNode {
+ unsigned int minRep=0;
+ unsigned int maxRep=0;
+ unsigned int nBonds=0;
+ std::vector<std::pair<unsigned int,unsigned int>> bondAtoms;
+};
+
+namespace utils {
+
+
+inline std::vector<LinkNode> getMolLinkNodes(const ROMol &mol, bool strict=true) {
+  std::vector<LinkNode> res;
+  std::string pval;
+  if (!mol.getPropIfPresent(common_properties::molFileLinkNodes, pval)) {
+    return res;
+  }
+  std::vector<int> mapping;
+
+  boost::char_separator<char> pipesep("|");
+  boost::char_separator<char> spacesep(" ");
+  for (auto linknodetext : tokenizer(pval, pipesep)) {
+    LinkNode node;
+    tokenizer tokens(linknodetext, spacesep);
+    std::vector<unsigned int> data;
+    try {
+      std::transform(tokens.begin(), tokens.end(), std::back_inserter(data),
+                     [](const std::string &token) -> unsigned int {
+                       return boost::lexical_cast<unsigned int>(token);
+                     });
+    } catch (boost::bad_lexical_cast &) {
+      std::ostringstream errout;
```

Comment:
Would this be better in a function?  There are 5 instances of pretty much the same code, with a different message.

Reply:
in principle I agree with this, but when I actually do so it becomes non-obvious that there's actually an exception thrown, so the gain in brevity is offset by a loss in clarity.